### PR TITLE
Ansible initiative

### LINKED ISSUE #14: [Help Needed] OpenSearch/Dashboards in Ansible
[Help Needed] OpenSearch/Dashboards in Ansible.
We invite communities to help us adding more distros of OpenSearch/Dashboards.
We recently have Helm OpenSearch available with the help of community, and we would love more community contributions.

### PR DESCRIPTION
**Pull request by saravanan30erd**
### Description
Ansible scripts for opensearch setup
 
### Issues Resolved
#14 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by saravanan30erd** [comment]
@peterzhuamazon 

I have split the task into many phases, currently I have completed the single node opensearch setup.

I will work on below tasks in upcoming days,
- opensearch multi-node cluster setup
- opensearch dashbaords Installation
- Performance analyzer plugin configuration

**Comment by saravanan30erd** [comment]
@peterzhuamazon 

Completed the task for `opensearch dashbaords Installation`. I think we can start here since pretty much I covered all which is necessary for single node installation.

File: ansible/inventories/opensearch/group_vars/all/all.yml
Comment (bbarani) [inline ansible/inventories/opensearch/group_vars/all/all.yml:24]: @saravanan30erd can we encapsulate the credentials rather than adding it directly to yml?

**Comment by peterzhuamazon** [comment]
@DandyDeveloper @TheAlgo Please review and comment when you have time.

**Comment by peterzhuamazon** [comment]
Adding @hyandell to co-proofread for any changes in the PR.

**Review by hyandell — COMMENTED** [review]
Hi Saravanan,

As Ansible is GPL licensed and has many plugins under different licenses, did you base your work here on a pre-existing codebase? The concern with code that touches on other projects being that we don't want to confuse the licensing with different licenses that affect the overall licensing.

I did see that you had https://github.com/saravanan30erd/opendistro_standalone_installation - which I assume this is based partly on; I'm just concerned about any other repos either this PR or that earlier repo is based on.

**Comment by saravanan30erd** [comment]
Hi @hyandell 

I have created this [opendistro](https://github.com/saravanan30erd/opendistro_standalone_installation) repo from scratch, its not based on any external repo. Also this PR is using only few logics from opendistro one but major areas are created from scratch.

File: ansible/inventories/opensearch/group_vars/all/all.yml
Comment (saravanan30erd) [reply ansible/inventories/opensearch/group_vars/all/all.yml]: ok

File: ansible/roles/centos7/dashboards/tasks/dashboards.yml
Comment (saravanan30erd) [reply ansible/roles/centos7/dashboards/tasks/dashboards.yml]: ok

**Comment by saravanan30erd** [comment]
@bbarani  I have made the changes based on your feedback.

**Comment by peterzhuamazon** [comment]
Need more eyes to check this @smlx let me know if you can help with the approval.
I will test this in a few days as I am currently working on another OpenSearch releases.

**Comment by hyandell** [comment]
> Hi @hyandell
> 
> I have created this [opendistro](https://github.com/saravanan30erd/opendistro_standalone_installation) repo from scratch, its not based on any external repo. Also this PR is using only few logics from opendistro one but major areas are created from scratch.

Thank you Saravanan :)

**Review by hyandell — APPROVED** [review]

**Comment by DandyDeveloper** [comment]
@peterzhuamazon I think for this one, I don't see any problems, but I don't have the time to test this at this moment in time. So I'll pass on this specific PR for review.

**Comment by peterzhuamazon** [comment]
Thanks @hyandell @DandyDeveloper for your inputs on this.
@saravanan30erd I will start the tests next week.
@TheAlgo please also take a look when you have time.

Thanks.

File: ansible/README.md
Comment (TheAlgo) [inline ansible/README.md:1]: Nit : OpenSearch with OpenSearch Dashboards

File: ansible/roles/centos7/dashboards/templates/opensearch_dashboards.yml
Comment (TheAlgo) [inline ansible/roles/centos7/dashboards/templates/opensearch_dashboards.yml:5]: Can we use a different name here , maybe `dashboardserver`?

File: ansible/roles/centos7/dashboards/vars/main.yml
Comment (TheAlgo) [inline ansible/roles/centos7/dashboards/vars/main.yml:2]: Can we name all instances of `elasticsearch` to `opensearch` and `kibana` to dashboards here?

**Review by TheAlgo — CHANGES_REQUESTED** [review]

**Comment by TheAlgo** [comment]
@saravanan30erd I took a quick pass. PR looks good to me except some nitpicks regarding naming. Good work !!

File: ansible/roles/centos7/dashboards/templates/opensearch_dashboards.yml
Comment (saravanan30erd) [reply ansible/roles/centos7/dashboards/templates/opensearch_dashboards.yml:5]: @TheAlgo  `kibanaserver` is the name used in default configuration file from Opensearch dashboards source file. Refer: https://opensearch.org/docs/dashboards/install/tls/ 

I suggest we should follow the default configuration from installation source to avoid confusions. May we it should be changed first in source files and official documentations then we can change it.

File: ansible/roles/centos7/dashboards/vars/main.yml
Comment (saravanan30erd) [reply ansible/roles/centos7/dashboards/vars/main.yml:2]: ok

Comment (saravanan30erd) [reply ansible/roles/centos7/dashboards/vars/main.yml]: @TheAlgo  Done the changes.

File: ansible/roles/centos7/dashboards/templates/opensearch_dashboards.yml
Comment (TheAlgo) [reply ansible/roles/centos7/dashboards/templates/opensearch_dashboards.yml:5]: @peterzhuamazon Just curious , why are we using `kibanaserver` till now?

Comment (saravanan30erd) [reply ansible/roles/centos7/dashboards/templates/opensearch_dashboards.yml:5]: @TheAlgo @peterzhuamazon `kibanaserver` is used in security plugin where it should be changed and then configuration. I raised the below PRs to fox this. Once this merged and released, then we can update here.

https://github.com/opensearch-project/security/pull/1443
https://github.com/opensearch-project/opensearch-build/pull/458

**Comment by peterzhuamazon** [comment]
https://github.com/opensearch-project/security/pull/1443

File: ansible/roles/centos7/dashboards/templates/opensearch_dashboards.yml
Comment (peterzhuamazon) [reply ansible/roles/centos7/dashboards/templates/opensearch_dashboards.yml:5]: We are still in discussion with Security Team on this @saravanan30erd please hold on for some time before we make a decision on this.

Thanks.

Comment (saravanan30erd) [reply ansible/roles/centos7/dashboards/templates/opensearch_dashboards.yml:5]: @peterzhuamazon  Sure, I think we can continue with `kibanaserver` here as of now.

**Comment by saravanan30erd** [comment]
@peterzhuamazon  I think we can rename `kibanaserver` in this project after its changed in security plugin and dashboard configuration, now we can continue the work here.

**Comment by peterzhuamazon** [comment]
> @peterzhuamazon I think we will rename `kibanaserver` later in this project after its changed in security plugin and dashboard configuration, now we can continue the work here.

I agree let us keep this in mind for now.
Can you create an issue to track this?
Thanks.

File: ansible/roles/centos7/dashboards/templates/opensearch_dashboards.yml
Comment (peterzhuamazon) [reply ansible/roles/centos7/dashboards/templates/opensearch_dashboards.yml:5]: We will keep this for now @TheAlgo.
@saravanan30erd please create an issue to track this.
Thanks.

**Comment by saravanan30erd** [comment]
@peterzhuamazon https://github.com/opensearch-project/opensearch-devops/issues/69

**Comment by peterzhuamazon** [comment]
Hi @saravanan30erd 

We have recently created a ansible-playbook repo here: https://github.com/opensearch-project/ansible-playbook

To avoid migration issues like helm-charts, would you mind move this PR to there as well?
We can link this PR as reference.

This would be easier for people to access Ansible as a whole without mixing with other components.
You just need to add a pointer from devops repo to ansible repo once complete, so people knows where is ansible playbook.

This devops repo will serve as a central hub for all the OpenSearch Project related automations.

Let me know what you think.

Thanks.

**Comment by saravanan30erd** [comment]
@peterzhuamazon  Great, I will move this PR to new repo. Its better to maintain the separate repo as it gets complex when we add other features.

**Comment by peterzhuamazon** [comment]
> @peterzhuamazon Great, I will move this PR to new repo. Its better to maintain the separate repo as it gets complex when we add other features.

That is exactly the reason why we make the standalone repo.
Just take a bit time since Ansible itself is not Apache2 to start with, so we carefully review the use case and your PR content before we proceed with the repo creation.
Thanks.

**Comment by peterzhuamazon** [comment]
Please close this PR once you raise the other one and link this one there.

**Comment by saravanan30erd** [comment]
@peterzhuamazon I moved this PR to new ansible repo(https://github.com/opensearch-project/ansible-playbook/pull/2).

**Comment by peterzhuamazon** [comment]
Thanks @saravanan30erd close this one and lets work on the other one, I migrate #69 to ansible playbook repo as well.
I also create an new issue here #71 so you can take action to create pointer to point people from devops to ansible repo.

Thanks.

**Pull request closed without merging**
